apps/login/lib/zitadel.ts

```diff
@@ -127,30 +135,36 @@ export type AddHumanUserData = {
   firstName: string;
   lastName: string;
   email: string;
-  password: string;
+  password: string | undefined;
 };
 
-export function addHumanUser(
+export async function addHumanUser(
   server: ZitadelServer,
   { email, firstName, lastName, password }: AddHumanUserData
 ): Promise<string> {
   const mgmt = user.getUser(server);
   return mgmt
     .addHumanUser(
-      {
-        email: { email },
-        username: email,
-        profile: { firstName, lastName },
-        password: { password },
-      },
+      password
+        ? {
+            email: { email },
+            username: email,
+            profile: { firstName, lastName },
+            password: { password },
+          }
+        : {
+            email: { email },
+            username: email,
+            profile: { firstName, lastName },
+          },
       {}
```

Comment:
let's use the spread operator to make it more DRY

Reply:
done